quantgov/estimator/structures.py

```diff
@@ -85,3 +113,41 @@ class CandidateModel(
             parameter values to test as values
     """
     pass
+
+
+class QGLdaModel(BaseEstimator, TransformerMixin):
+    @check_gensim
+    @check_spacy
+    def __init__(self, word_regex=r'\b[A-z]{2,}\b', stop_words=STOP_WORDS):
+        self.stop_words = stop_words
+        self.word_regex = re.compile(word_regex)
+
+    def transform(self, driver):
+        self.test_corpus = self.create_corpus(driver)
+        return self.model.transform(self.test_corpus)
+
+    def create_corpus(self, driver):
+        return [self.dictionary.doc2bow([i.group(0).lower()
+                for i in self.word_regex.finditer(doc.text)])
+                for doc in driver.stream()]
+
+    def fit(self, driver, alpha=None, eta=None, num_topics=1, passes=1):
+        self.dictionary = Dictionary([[i.group(0).lower()
+                                      for i in self.word_regex
+                                        .finditer(doc.text)]
+                                      for doc in driver.stream()])
+        stop_ids = [self.dictionary.token2id[stopword] for stopword
```

Comment:
Wouldn't it be better to only pass the dictionary words that aren't in stop_words?